cernopendata/jsonschemas/records/record-v1.0.0.json

```diff
@@ -480,6 +480,13 @@
       },
       "type": "array"
     },
+    "signature": {
+      "items": {
+        "description": "The final state particles in a dataset (e.g. electron)",
+        "type": "string"
+      },
+      "type": "array"
+    },
```

Comment:
The facet works well, however the commit also introduces new records for files that already belong to other records, so our integrity checks fail with errors like:

```console
$ docker exec -i -t opendatacernch_web_1 /code/scripts/populate-instance.sh
...
Recid 545 file Wmunu.csv could not be loaded due to (psycopg2.IntegrityError) duplicate key value violates unique constraint "uq_files_files_uri"
DETAIL:  Key (uri)=(root://eospublic.cern.ch//eos/opendata/cms/Run2011A/SingleMu/CSV/12Oct2013-v1/Wmunu.csv) already exists.
 [SQL: 'INSERT INTO files_files (created, updated, id, uri, storage_class, size, checksum, readable, writable, last_check_at, last_check) VALUES (%(created)s, %(updated)s, %(id)s, %(uri)s, %(storage_class)s, %(size)s, %(checksum)s, %(readable)s,
%(writable)s, %(last_check_at)s, %(last_check)s)'] [parameters: {'updated': datetime.datetime(2020, 8, 18, 9, 18, 21, 778499), 'created': datetime.datetime(2020, 8, 18, 9, 18, 21, 778494), 'checksum': u'adler32:0607d5ee', 'last_check_at': None, 'readable': True, 'uri': u'root://eospublic.cern.ch//eos/opendata/cms/Run2011A/SingleMu/CSV/12Oct2013-v1/Wmunu.csv', 'last_check': True, 'writable': False, 'storage_class': 'S', 'id': UUID('61c57edd-44ab-4071-9c1a-b83b3266cb46'), 'size': 7969331}]
```

The duplicate files are:

```console
$ cat cernopendata/modules/fixtures/data/records/*.json | jq -S '.[].files | .[].uri' | sort | uniq -D
"root://eospublic.cern.ch//eos/opendata/cms/masterclass-2014/4lepton.csv"
"root://eospublic.cern.ch//eos/opendata/cms/masterclass-2014/4lepton.csv"
"root://eospublic.cern.ch//eos/opendata/cms/masterclass-2014/diphoton.csv"
"root://eospublic.cern.ch//eos/opendata/cms/masterclass-2014/diphoton.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/CSV/12Oct2013-v1/Zee.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/CSV/12Oct2013-v1/Zee.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleMu/CSV/12Oct2013-v1/Dimuon_DoubleMu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleMu/CSV/12Oct2013-v1/Dimuon_DoubleMu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleMu/CSV/12Oct2013-v1/Jpsimumu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleMu/CSV/12Oct2013-v1/Jpsimumu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleMu/CSV/12Oct2013-v1/Ymumu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleMu/CSV/12Oct2013-v1/Ymumu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleMu/CSV/12Oct2013-v1/Zmumu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleMu/CSV/12Oct2013-v1/Zmumu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/SingleElectron/CSV/12Oct2013-v1/Wenu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/SingleElectron/CSV/12Oct2013-v1/Wenu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/SingleMu/CSV/12Oct2013-v1/Dimuon_SingleMu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/SingleMu/CSV/12Oct2013-v1/Dimuon_SingleMu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/SingleMu/CSV/12Oct2013-v1/Wmunu.csv"
"root://eospublic.cern.ch//eos/opendata/cms/Run2011A/SingleMu/CSV/12Oct2013-v1/Wmunu.csv"
```

Two possibilities how to address this problem:

- We copy the files into a new location at the price of duplicating them.  The sizes of CSV files are small, so this could be acceptable.

- We "unlink" files from existing records [such as this record](http://opendata.cern.ch/record/545) that would not show any files anymore -- instead it would show links to the new records where individual files can be downloaded.

@katilp Which solution do you prefer?

Reply:
No strong opinions, there may be links to the files directly from notebooks such as 
`data = pd.read_csv('http://cern.ch/opendata/record/545/files/Dimuon_DoubleMu.csv')`
Is it easier for one of the solution above to keep them valid?

Reply:
If that's possible for the second option (i.e. keep records like 545 as an umbrella record) so that the files would be available both from `http://cern.ch/opendata/record/545/files/` (for backward compatibility) and `http://cern.ch/opendata/record/[newrecordnumber]/files/` but they would only show under the new record number on the portal

Reply:
To keep the old links active, it is easier to just use the first option and copy the files to new locations. 

Alternatively, we could also *not* attach files to the new records, but keep them attached to the old record, and link the files from the new records to the old one.  So that the new records won't have "Files" section, but something like "URLs" section from where the files can be downloaded... A bit like "Source code repository" links in http://opendata.cern.ch/record/550. This option would be the easiest, as there are no files to duplicate...

Reply:
OK, then the first option is good. It is probably best practice to have the files attached to the new records, because that's where people would expect them to appear. That's more logical for the data files, while for the sw records the source code type solution is natural. But again, no strong opinions.

Reply:
Any preference as to the naming of new file locations? If not, we could simply use something neutral such as "../files/..." subdirectory, giving:

- old: `/eos/opendata/cms/masterclass-2014/dielectron.json`
- new: `/eos/opendata/cms/masterclass-2014/files/dielectron.json`

and:

- old: `/eos/opendata/cms/Run2011A/DoubleMu/CSV/12Oct2013-v1/Jpsimumu.csv`
- new: `/eos/opendata/cms/Run2011A/DoubleMu/CSV/12Oct2013-v1/files/Jpsimumu.csv`

Reply:
OK, in the old one for the Jpsimumu file, the processing tag `12Oct2013-v1`  is a bit of overdoing as the part `Run2011A/DoubleMu` already nicely indicates from which dataset the derived file comes from. 
Or are there different reprocessings? The old csv's that were made public  before the first release may possible have different parent dataset.

Reply:
Now, I see, it is there to be similar to the AOD file path. What do you think? We can keep it if you like, it is not necessary but does not harm either.

Reply:
Yeah, it is easier to keep the old convention (similar to AODs) and just add a new subdirectory named like `files` for the new individual records' copies.  If this is OK with you, I'll amend PR in this sense